src/server/status_server.rs

```diff
@@ -59,19 +64,75 @@ impl StatusServer {
         let builder = Server::try_bind(&addr)?;
 
         // Create a status service.
-        let service = |req: Request<Body>| -> FutureResult<Response<Body>, hyper::Error> {
-            let mut response = Response::new(Body::empty());
+        let service = |req: Request<Body>| -> Box<dyn Future<Item=Response<Body>, Error=hyper::Error> + Send> {
+            let path = req.uri().path().to_owned();
+            let method = req.method().to_owned();
+            let on_failpoints_path: bool = path.starts_with(FAIL_POINTS_PATH);
+
+            match (method, path.as_ref()) {
+                (Method::GET, "/metrics") => Box::new(ok(Response::new(dump().into()))),
+                (Method::GET, "/status") => Box::new(ok(Response::default())),
+
+                // Below are fail points related endpoints
+                (Method::PUT, _) if on_failpoints_path => {
```

Comment:
can we move all associated fail operations to a function?

Reply:
I can give it a try